abis_mapping/base/mapper.py

```diff
@@ -66,6 +66,76 @@ def apply_mapping(
             rdflib.Graph: ABIS Conformant RDF Sub-Graph from Raw Data Chunk.
         """
 
+    @final
+    @classmethod
+    def extract_extra_fields(
+        cls,
+        row: frictionless.Row
+    ) -> dict[str, Any]:
+        """Extracts extra values from a row not in template schema.
+
+        Args:
+            row (frictionless.Row): Row of data including extra rows.
+
+        Returns:
+            dict[str, Any]: Dictionary containing extra values, if any.
+        """
+        # Get schema consisting of extra fields
+        extra_schema = cls.extra_fields_schema(row)
+
+        # Create dictionary consisting row data from extra fields only
+        return {field: row[field] for field in extra_schema.field_names if row[field] is not None}
+
+    @final
+    @classmethod
+    def extra_fields_schema(
+        cls,
+        data: types.ReadableType | frictionless.Row
```

Comment:
Ha! This new union type notation (`|` instead `Union[..]` )  is python 3.10 and above only. We are still stuck with python 3.9 atm with BDR-partA.  
I would recommend to forgo `types.ReadableType` as an acceptable type for now, see comment below.

Reply:
Removed the ReadableType method overloading. See comment above or below about potential future optimisation.